includes/payment-methods/class-wc-stripe-upe-payment-gateway.php

```diff
@@ -392,6 +397,9 @@ public function get_upe_enabled_at_checkout_payment_method_ids( $order_id = null
 			if ( ! $is_automatic_capture_enabled && $method->requires_automatic_capture() ) {
 				continue;
 			}
+			if ( ! $this->payment_methods[ $payment_method_id ]::is_available() ) {
+				continue;
+			}
```

Comment:
IMO we could remove this check to reduce the noise, since ideally if Link is not available for a given country, it won't even be enabled for checkout in the first place.

So, in case the account is changed, #2468 will reset any enabled payment methods that aren't supposed to be available.

In case you disagree with the above or I'm missing something, another suggestion here would be to reuse `$method` instead of  `$this->payment_methods[ $payment_method_id ]`.

Reply:
changed in [0ea817f](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2452/commits/0ea817f020274f678061c27bbd199a874a2578e8) . thx